UKHomeOffice/cop-ui
# COP-8541 Add filtering for targets by group

## AC

- /targets should show users a list of all their groups and a count of targets for those groups
- /targets should default to a list of targets for all groups
- if user checks a group (or multiple groups) and clicks apply filters the list of targets should refresh to show only targets in those group(s)
- if user unchecks a group (or multiple groups) and clicks apply filters the list of targets should refresh to show only targets in those group(s) that remain checked
- pagination should persist group selections and not reset groups
- if user clicks clear all the list of targets should refresh to show targets for all groups again
- group list should be sorted alphabetically

## Updated

- added groups filter list
- added apply and clear all filter actions

## To test

- go to /targets
- > you should see all targets for your groups
- > you should see a list of groups on the left
- > groups should be sorted alphabetically
- tick a group on the left that has more than 20 targets
- click apply filter
- > you should see a reduction in the number of pages in the pagination (related to how many targets there are)
- click next on pagination
- > you should see page 2 of the targets for your selected group
- click clear all filters
- > you should see all the targets again
